app/scripts/views/sign_up.js

```diff
@@ -314,6 +315,7 @@ function (Cocktail, p, BaseView, FormView, Template, AuthErrors, mailcheck,
 
           if (preVerifyToken && account.get('verified')) {
             self.logViewEvent('preverified.success');
+            self.onSignInSuccess(account);
```

Comment:
These are a lot of initial thoughts, but I am pro moving more towards events because it seems like it could make a lot of the logic less procedural and more decoupled. Plus those new events could be used for other uses when we find htem.